Index: carmen-0.7.4-beta/src/base/scoutlib/Ndirect.c
===================================================================
--- carmen-0.7.4-beta.orig/src/base/scoutlib/Ndirect.c	2010-08-05 20:02:17.176959918 +0200
+++ carmen-0.7.4-beta/src/base/scoutlib/Ndirect.c	2010-08-05 20:06:11.801913999 +0200
@@ -3532,7 +3532,12 @@
   tempp = (unsigned long long *)&data;
   *tempp = temp;
   
-  return(data);
+  /* gcc fails here with the error: 'data' is used uninitialized in this
+    function. Since data is actually uninitalized, extractdouble() isn't
+    called anywhere in the code, I'm returning 0 here just to keep gcc
+    going on */
+  /* return(data); */
+  return 0;
 }
 
 /************************************************
